### PR TITLE
TRM-23456 Make the public fields private and introduce accessor and mutator

### PR DESCRIPTION
**Pull request by sahmad-ebi**
Hi Leo,
Can you please review the changes? I have made the fields private and added getter and setter for them

### CODE REVIEW

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java [hunk 1]
```diff
@@ -601,9 +614,13 @@ public void exitCc_subcellular_location(CcLineParser.Cc_subcellular_locationCont
                             ccSubcellularLocationValueContext =
                                     locEvi.cc_subcellular_location_value();
 
-                    locationObject.orientation = new CcLineObject.LocationValue();
-                    locationObject.orientation.value =
-                            ccSubcellularLocationValueContext.cc_subcellular_words().getText();
+                    locationObject.setOrientation(new LocationValue());
```
Comment (LeonardoGonzales) [inline ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java]: I think the code would be easier to read if you create the LocationValue, set the orientation value and after that you set it to the locationObject. What do you think?

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java [hunk 2]
```diff
@@ -714,7 +733,7 @@ public void exitCc_alternative_products(CcLineParser.Cc_alternative_productsCont
                                 .CC_AP_FEATURE_IDENTIFIER();
                 for (TerminalNode terminalNode : terminalNodes) {
                     String text = terminalNode.getText();
-                    name.sequenceFTId.add(text);
+                    name.getSequenceFTId().add(text);
```
Comment (LeonardoGonzales) [inline ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java]: This is optional, Do you think would be easier to read the code if you create the list, add the items.. and at the end set it to an object?  This is a general comment... I saw it in many other classes as well..

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java [hunk 3]
```diff
@@ -880,22 +900,22 @@ public void exitCc_catalytic_activity(CcLineParser.Cc_catalytic_activityContext
         if (lineCtx != null) {
             List<Cc_cat_act_pd_lineContext> lines = lineCtx.cc_cat_act_pd_line();
             if (!lines.isEmpty()) {
-                dd.physiologicalDirections = new ArrayList<>();
+                dd.setPhysiologicalDirections(new ArrayList<>());
```
Comment (LeonardoGonzales) [inline ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java]: just create a list here..... and set it after the for....

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/cclineobject/CcLineObject.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    @Override
+    public EvidenceInfo getEvidenceInfo() {
+        return evidenceInfo;
+    }
+
+    public void setEvidenceInfo(EvidenceInfo evidenceInfo) {
+        this.evidenceInfo = evidenceInfo;
+    }
+
+    public enum AlternativeNameSequenceEnum {
```
Comment (LeonardoGonzales) [inline ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/cclineobject/CcLineObject.java]: Maybe these enums could go inside the class that is using it... for example, AlternativeNameSequenceEnum could go inside AlternativeProductName.. is it possible?

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/cclineobject/EvidencedString.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public static EvidencedString get(String text1, List<String> strings) {
+        if (strings == null || strings.isEmpty()) {
+            return get(text1);
+        } else {
+            return new EvidencedString(text1, strings);
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (LeonardoGonzales) [inline ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/cclineobject/EvidencedString.java]: Equals and Hash for the others as well?

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/ft/FtLineObject.java [hunk 6]
```diff
@@ -24,15 +24,63 @@ public EvidenceInfo getEvidenceInfo() {
     }
 
     public static class FT {
-        public FTType type;
+        private FTType type;
 
         // can have fuzzy type thus cannot use int.
-        public String location_start;
-        public String location_end;
+        private String location_start;
```
Comment (LeonardoGonzales) [inline ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/ft/FtLineObject.java]: not part of the task, so, very very very optional, is it possible to change it to camelCase?

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java (hunk 2 above)
Comment (sahmad-ebi) [reply ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java]: I'd say this is more like style of coding. We are not used to it. 
If you replace this
name.getSequenceFTId().add(text);
with
List<String> ftids = new ArrayList<>();
ftids.add(text)
name.setSequenceFtId(ftids);
We may override the existing data in the list. Then we need to find out where we are adding object to the list and add another method like addSequenceFtId(String text) which will be implemented something like
public void addSequenceFtId(String text){
 if(this.sequenceFtid == null){
this.sequenceFtId = new ArrayList<>();
}
this.sequenceFtId.add(text);
}
I am not sure, if such change is needed. It is risky, i think.

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/cclineobject/EvidencedString.java (hunk 5 above)
Comment (sahmad-ebi) [reply ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/cclineobject/EvidencedString.java]: Those methods were already there. I just put it in a top level class.

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java (hunk 2 above)
Comment (LeonardoGonzales) [reply ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java]: We have few safe use cases, like the cofactor example

dd.cofactors = new ArrayList<>();	                dd.setCofactors(new ArrayList<>());
                for (Cc_cofactor_lineContext line : lines) {	                for (Cc_cofactor_lineContext line : lines) {
                    CcLineObject.CofactorItem item = new CcLineObject.CofactorItem();	                    CofactorItem item = new CofactorItem();
                    item.name = line.cc_cofactor_name().cc_properties_text_level2().getText();	                    item.setName(line.cc_cofactor_name().cc_properties_text_level2().getText());
                    item.xref = line.cc_cofactor_xref().cc_properties_text_level2().getText();	                    item.setXref(line.cc_cofactor_xref().cc_properties_text_level2().getText());
                    CcLineParser.Cc_cofactor_evidenceContext evidenceContext =	                    CcLineParser.Cc_cofactor_evidenceContext evidenceContext =
                            line.cc_cofactor_evidence();	                            line.cc_cofactor_evidence();
                    if (evidenceContext != null) {	                    if (evidenceContext != null) {
                        EvidenceInfo.processEvidence(	                        EvidenceInfo.processEvidence(
                                object.getEvidenceInfo(), item, evidenceContext.EV_TAG());	                                object.getEvidenceInfo(), item, evidenceContext.EV_TAG());
                    }	                    }
                    dd.cofactors.add(item);	                    dd.getCofactors().add(item);
                }

Comment (sahmad-ebi) [reply ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/cc/CcLineModelListener.java]: ok. I am doing the simple ones.

File: ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/ft/FtLineObject.java (hunk 6 above)
Comment (sahmad-ebi) [reply ff-parser/src/main/java/org/uniprot/core/flatfile/parser/impl/ft/FtLineObject.java]: https://www.ebi.ac.uk/panda/jira/browse/TRM-23567

**Comment by LeonardoGonzales** [comment]
Looks excellent!!

Tks!!

**Pull request merged**